### PR TITLE
[gitpod-db] Add region for cluster selection

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
DB migration. adding a region column to `d_b_workspace_cluster`

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`
- [ ] /werft publish-to-npm

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/gitpod-db/src/typeorm/migration/1675694207999-WorkspaceClusterRegion.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { columnExists } from "./helper/helper";
+
+const table = "d_b_workspace_cluster";
+const column = "region";
+
+export class WorkspaceClusterRegion1675694207999 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, table, column))) {
+            await queryRunner.query(
+                `ALTER TABLE ${table} ADD COLUMN ${column} varchar(255) NOT NULL DEFAULT '', ALGORITHM=INPLACE, LOCK=NONE`,
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/migration/1675694207999-WorkspaceClusterRegion.ts:17]: Do you need an index on this also?

File: components/gitpod-db/src/typeorm/migration/1675694207999-WorkspaceClusterRegion.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            );
+        }
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        if (await columnExists(queryRunner, table, column)) {
+            await queryRunner.query(`ALTER TABLE ${table} DROP COLUMN ${column}`);
```
Comment (easyCZ) [inline components/gitpod-db/src/typeorm/migration/1675694207999-WorkspaceClusterRegion.ts:24]:
```suggestion
            await queryRunner.query(`ALTER TABLE ${table} DROP COLUMN ${column}, ALGORITHM=INPLACE, LOCK=NONE`);
```

**Review by easyCZ — APPROVED** [review]

**Comment by svenefftinge** [comment]
No index neccessary given the table is rather small (has 10 entries in prod)

/unhold

**Pull request merged**